fig/service.py

```diff
@@ -629,6 +634,25 @@ def split_port(port):
     return internal_port, (external_ip, external_port or None)
 
 
+def build_extra_hosts(extra_hosts_config):
+    if extra_hosts_config is None:
+        return None
+
+    if isinstance(extra_hosts_config, basestring):
+        extra_hosts_config = [extra_hosts_config]
+
+    extra_hosts_dict = {}
+    for extra_hosts_line in extra_hosts_config:
+        if isinstance(extra_hosts_line, dict):
+            # already interpreted as a dict (depends on pyyaml version)
+            extra_hosts_dict.update(extra_hosts_line)
+        else:
+            # not already interpreted as a dict
+            extra_hosts_dict.update(extra_hosts_line.split(':'))
```

Comment:
I believe line 651 should read:

    extra_hosts_dict.update([extra_hosts_line.split(':')])